File: lib_openweather/src/main/java/guepardoapps/library/openweather/controller/INetworkController.java
package guepardoapps.library.openweather.controller;

import android.support.annotation.NonNull;

@SuppressWarnings({"unused"})
public interface INetworkController {
    String WIFIReceiverInHomeNetworkBroadcast = "guepardoapps.library.openweather.controller.wifi.home_network.yes";
    String WIFIReceiverNoHomeNetworkBroadcast = "guepardoapps.library.openweather.controller.wifi.home_network.no";

    boolean IsNetworkAvailable();

    boolean IsWifiConnected();

    boolean IsMobileDataEnabled();

    boolean IsNetworkTypeEnabled(int networkType);

    boolean IsHomeNetwork(@NonNull String homeSSID);

    boolean SetWifiState(boolean newWifiState);

    boolean SetMobileDataState(boolean newMobileDataState);

    String GetWifiSsid();

    String GetIpAddress();

    int GetWifiDBM();
}

File: lib/src/main/java/guepardoapps/lib/openweather/controller/INetworkController.kt
package guepardoapps.lib.openweather.controller

import android.net.NetworkInfo

internal interface INetworkController {
    fun isInternetConnected(): Pair<NetworkInfo?, Boolean>
    fun getIpAddress(): String

    fun isWifiConnected(): Pair<NetworkInfo?, Boolean>
    fun isHomeWifiConnected(ssid: String): Boolean
    fun getWifiSsid(): String
    fun getWifiDBM(): Int
    fun setWifiState(newWifiState: Boolean)

    fun isMobileConnected(): Pair<NetworkInfo?, Boolean>
    fun setMobileDataState(newMobileDataState: Boolean)
}